sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -370,8 +370,17 @@ abstract class CSVSuite
           .load(testFile(carsFile)).collect()
       }
 
-      assert(exception.getMessage.contains("Malformed CSV record"))
-      assert(ExceptionUtils.getRootCause(exception).isInstanceOf[RuntimeException])
+      checkError(
+        exception = exception.getCause.asInstanceOf[SparkException],
+        errorClass = "_LEGACY_ERROR_TEMP_2177",
+        parameters = Map("failFastMode" -> "FAILFAST")
+      )
```

Comment:
This fails on `CSVv2Suite` since it raises `_LEGACY_ERROR_TEMP_2064` instead of `_LEGACY_ERROR_TEMP_2177`.

Is there any way we separate the tests between CSVv1 and CSVv2 ??

Reply:
You could move the check to those specific test suites.